### DISCUSSION TITLE
WrapPanel in ListBox doesn't wrap but shows scrollbar

### DISCUSSION THREAD

**Discussion by PeterVD-Lansweeper** [AvaloniaUI/Avalonia#5517] [Q&A] [answered]
Hi,

I'm trying to use a WrapPanel as the ItemsPanel in a ListBox, but I don't get the behavior I want. (I'm using Avalonia v0.10.0)
The WrapPanel doesn't wrap, but puts all items on one line and shows a horizontal scrollbar.

This is the XAML:
```
<ListBox Items="{Binding CodeOwners, Mode=OneWay}" >
	<ListBox.ItemsPanel>
		<ItemsPanelTemplate>
			<WrapPanel />
		</ItemsPanelTemplate>
	</ListBox.ItemsPanel>
	<ListBox.ItemTemplate>
		<DataTemplate>
			<StackPanel Orientation="Horizontal">
				<CheckBox IsChecked="{Binding IsSelected, Mode=TwoWay}"></CheckBox>
				<TextBlock Text="{Binding Description, Mode=OneWay}" Margin="5,0,0,0" ></TextBlock>
			</StackPanel>
		</DataTemplate>
	</ListBox.ItemTemplate>
</ListBox>
```

What am I doing wrong here? 
I tried binding the WrapPanel's Width to the width of the ListBox, but that didn't have any effect.

When I give the ListBox a fixed size, the wrapping works fine. 
But using a Grid-Layout and let it take al the space that is available (in a ColumnDefinition Width="*") and the WrapPanel's size is messed up.

**Comment by jp2masa** [comment, marked as answer]
It works for me: [ListBoxBinding.zip](https://github.com/AvaloniaUI/Avalonia/files/6011267/ListBoxBinding.zip).

Probably some `ListBox` parent is measuring with infinite width (e.g. a `StackPanel`, `ScrollViewer`,...).

**Reply by FoggyFinder** [reply]
Try to add this

```xml
ScrollViewer.HorizontalScrollBarVisibility="Disabled"
```
i.e.

```xml
<ListBox Items="{Binding CodeOwners, Mode=OneWay}" ScrollViewer.HorizontalScrollBarVisibility="Disabled">
```

**Reply by PeterVD-Lansweeper** [reply]
That did the trick. Thanks. :-)
